src/sentry/notifications/helpers.py

```diff
@@ -981,8 +990,46 @@ def get_setting_providers_for_projects(
     return result
 
 
-def get_setting_providers_for_users(
-    user_ids: Iterable[int],
+def get_enabled_notification_settings_by_recipients(
```

Comment:
We should be able to abstract this logic of checking both tables for enabled notification types and providers and avoid repeating the logic in various parts of the codebase.